benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java

```diff
@@ -66,7 +74,7 @@
     private final LongAdder totalMessagesSent = new LongAdder();
     private final LongAdder totalMessagesReceived = new LongAdder();
 
-    private boolean testCompleted = false;
+    private volatile boolean testCompleted = false;
```

Comment:
I had deliberately left the field to be non-volatile, and the reason was to avoid the volatile check at each iteration, since we do not require strict happened-before semantic, rather just a way to have the test stopped. In practice, with the dirty read the producing thread will "see" the change almost immediately.

Reply:
I would agree with you in this case there is no reason to use volatile here.